src/spaces/discrete.jl

```diff
@@ -117,6 +117,21 @@ function random_empty(model::ABM{<:DiscreteSpace}, cutoff = 0.998)
     end
 end
 
+"""
+    empty_nearby_positions(position, model::ABM{<:DiscreteSpace}, r = 1; kwargs)
+    empty_nearby_positions(agent, model::ABM{<:DiscreteSpace}, r = 1; kwargs)
+
+Return an iterable of all empty positions within "radius" `r` of the given `position` or `agent`.
+
+The value of r and possible keywords operate identically to `nearby_positions`.
+
+This function only exists for discrete spaces with a finite amount of positions.
+"""
+empty_nearby_positions(agent::A, model, r = 1) where {A<:AbstractAgent} = 
+    empty_nearby_positions(agent.pos, model, r)
+empty_nearby_positions(pos, model, r = 1) = filter(x -> isempty(x, model), collect(nearby_positions(pos, model, r)))
```

Comment:
Is this `collect` needed?

Reply:
`collect` is not needed. However you need to replace `filter` with `Iterators.filter` to allow filtering an arbitrary iterable like the output of `nearby_positions`.